assets/js/base/context/cart-checkout/checkout/processor/index.js

```diff
@@ -13,6 +13,26 @@ import {
 import { useEffect, useRef, useCallback } from '@wordpress/element';
 import { useStoreNotices } from '@woocommerce/base-hooks';
 
+/**
+ * @typedef {import('@woocommerce/type-defs/payments').PaymentDataResponse} PaymentDataResponse
+ */
+
+/**
+ * Utility function for preparing payment data for the request.
+ *
+ * @param {Object} paymentData Arbitrary payment data provided by the payment
+ *                             method.
+ *
+ * @return {PaymentDataResponse[]} Returns the payment data as an array of
+ *                                 PaymentDataResponse objects.
+ */
+const preparePaymentData = ( paymentData ) => {
```

Comment:
On the fence about whether this adds value as a separate function, since it's so generic - i.e. mapping to `[ { key:, value: }]` format. There's probably a lodash routine that does exactly this; I don't think this is customised for our payment use case (apart from the name & typedef). Just a thought - happy for this to merge as is :)

Reply:
Even if there was a lodash function for this (which I'm not aware of their being - did you know of one?) - what value is there introducing a dependency here on lodash just for this specific use case?

> I don't think this is customised for our payment use case (apart from the name & typedef).

I'm not sure what you mean by this exactly, but it _is_ specifically customized for payment use case because the `checkout` processing endpoint [_requires_ payment data to be sent in the body in this format](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/501296234d1903eb9df2645a57df602175af4616/src/RestApi/StoreApi/Routes/Checkout.php#L60-L80).